### DISCUSSION TITLE
Laser engraving speed

### DISCUSSION THREAD

**Discussion by DevEvV** [winder/Universal-G-Code-Sender#1556] [Help]
Why does the old version of the program work many times faster than the new one?
Will UGS be as incredibly fast in new versions as before? For fast laser engraving, baudrate 230400 is the limiting factor. It will be cool if it becomes possible to set higher values ​​(460800, 921600).

Thanks for your work, UGS is a great program!

![UGS](https://user-images.githubusercontent.com/78816441/111052591-39259c80-846d-11eb-82bc-12700c0b7340.jpg)

**Comment by breiler** [comment]
Thanks for the really clear comparison! May we have the sample program to test with? 
I have some ideas on what has changed in the code that could caused this.

**Reply by DevEvV** [reply]
This problem appears whenever you need to send a large number of commands through the serial port. Raster engraving is an example of such a task. I am using Grbl ESP32 controller.

![Test](https://user-images.githubusercontent.com/78816441/111079317-94599c80-850a-11eb-9d1c-d3f26bbb9897.jpg)
![Test 2](https://user-images.githubusercontent.com/78816441/111079332-9f143180-850a-11eb-872e-28f802d19907.jpg)


[TEST.zip](https://github.com/winder/Universal-G-Code-Sender/files/6137277/TEST.zip)
